modules/c++/mem/tests/ScratchVisualization.cpp

```diff
@@ -0,0 +1,477 @@
+#include <mem/ScratchMemory.h>
+#include <mem/BufferView.h>
+#include <cli/ArgumentParser.h>
+#include <str/Convert.h>
+#include <vector>
+#include <fstream>
+#include <string>
+
+struct Operation
```

Comment:
Is this now duplicated here and in the other test code? It would be good to move this to a common location so that the unittests and this both run the same operations. Otherwise things may get out of sync.

Reply:
What would be the best way to organize this? To put it as a header file in tests since the struct is only used for testing?